### PR TITLE
Add ability to initialize RMM with multiple GPUs

### PR DESCRIPTION
**Pull request by jrhemstad**
Depends on https://github.com/rapidsai/rmm/pull/146

Adds new `std::vector<int>` member to `rmmOptions_t`  for list of GPU device IDs to register with RMM. 

It will be the user's responsibility to call `cudaSetDevice(i)` before calling `RMM_ALLOC`/`RMM_FREE` to the appropriate device that was specified in `devices`.

If `devices.empty()`, it is assumed only a single GPU is used, which is determined by `cudaGetDevice()` (i.e., this will preserve the original single GPU behavior of RMM). This is the default case.

### CODE REVIEW

**Comment by harrism** [comment]
Depends on #132

**Comment by harrism** [comment]
@jrhemstad can you take a look at resolving the conflicts so we can get this in to 0.10?

**Comment by kkraus14** [comment]
@jrhemstad let me know when you need help with the Cython

**Comment by harrism** [comment]
@kkraus14 probably ready for Cython here.

**Comment by harrism** [comment]
@rerun tests

**Comment by harrism** [comment]
@jrhemstad failures are linker errors failing to find rmmGetAllocationOffset(). Could that be somehow related to the removal of `extern "C"`?

**Comment by jakirkham** [comment]
cc @pentschev (in case this is of interest)

File: python/rmm/_lib/lib.pxd [hunk 1]
```diff
@@ -65,10 +66,11 @@ cdef extern from "rmm/rmm.h" nogil:
         PoolAllocation = 1,
         CudaManagedMemory = 2,
 
-    ctypedef struct rmmOptions_t:
+    cdef cppclass rmmOptions_t:
```
Comment (shwina) [inline python/rmm/_lib/lib.pxd]: Needs to be declared a `cppclass` so that members are initialized with default constructors

**Comment by shwina** [comment]
rerun tests

**Comment by harrism** [comment]
Jake and I discussed we should delay this to 0.11.

**Comment by shwina** [comment]
rerun tests

**Comment by jrhemstad** [comment]
rerun tests

File: include/rmm/mr/cnmem_memory_resource.hpp [hunk 2]
```diff
@@ -55,41 +58,47 @@ class cnmem_memory_resource : public device_memory_resource {
...
-      throw std::runtime_error{"Failed to get CUDA device"};
+  cnmem_memory_resource(std::size_t initial_pool_size,
+                        std::vector<int> const& devices,
+                        memory_kind pool_type) {
+    std::vector<cnmemDevice_t> cnmem_devices;
+
+    // If no devices were specified, use the current one
+    if (devices.empty()) {
+      int current_device{};
+      cudaGetDevice(&current_device);
```
Comment (harrism) [inline include/rmm/mr/cnmem_memory_resource.hpp]: Check errors? Can we throw here?

File: include/rmm/mr/default_memory_resource.hpp [hunk 3]
```diff
@@ -17,6 +17,8 @@
 #pragma once
 
 #include "device_memory_resource.hpp"
+
+#include <vector>
```
Comment (harrism) [inline include/rmm/mr/default_memory_resource.hpp]: Why is this needed here?

Comment (jrhemstad) [reply include/rmm/mr/default_memory_resource.hpp]: Good question. It's not.
